### PR TITLE
Native course details

### PR DESCRIPTION
**Pull request by vargaat**
refs: MBL-14927
affects: Student, Teacher
release note: none

test plan:
- New selection indicator.
- Selection indicator is permanent in case of home and syllabus cells in split view.
- Going back to course details from a sub menu now restores the default home view.

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add only one affects.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add only one release note.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/CoreTests/Courses/CourseDetails/Model/TabFilterTests.swift:126:88: Trailing Comma Violation: Multi-line collection literals should have trailing commas. (trailing_comma)
    
    
</details>
<details><summary>:x: XCTest failed: StudentUnitTests/RoutesTests/testRoutesWithK5Alternative</summary>


    Assertion Failure at RoutesTests.swift:99: XCTAssertEqual failed: ("nil") is not equal to ("Optional("/courses/:courseID")")
</details>


[MBL-14927](https://instructure.atlassian.net/browse/MBL-14927)

#### Affected Apps: Student, Teacher
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/app/88a3641e29e4745c/build/00fd3988-d87c-4667-bca5-0230afcaef9d/artifact/be728261a5785581/p/a36ec5049837ac8409774bc4dd2b6dcd">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/app/88a3641e29e4745c/build/00fd3988-d87c-4667-bca5-0230afcaef9d/artifact/b5142c6383d2c0ff/p/a282b079584adb7d598d0e7b257d0b1f">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/app/88a3641e29e4745c/build/00fd3988-d87c-4667-bca5-0230afcaef9d/artifact/e3ccbcb9a5c8ed2c/p/235e1b60b77c048dc7e5cdf19fff7390">
    
  </a>
</details>

**Comment by balintbartok** [comment]
Course image is blank on iPad 14.6

File: Core/Core/Courses/CourseDetails/ViewModel/CourseSettingsViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    @Published public private(set) var hideColorOverlay: Bool?
+
+    @Environment(\.appEnvironment) private var env
+    private var isFirstAppearance = true
+    private var context: Context
+    private lazy var colors = env.subscribe(GetCustomColors())
+    private lazy var course = env.subscribe(GetCourse(courseID: context.id)) { [weak self] in
+        self?.courseDidUpdate()
+    }
+    private lazy var settings: Store<GetUserSettings> = env.subscribe(GetUserSettings(userID: "self")) { [weak self] in
```
Comment (balintbartok) [inline Core/Core/Courses/CourseDetails/ViewModel/CourseSettingsViewModel.swift:45]: missing linebreak

**Review by balintbartok — APPROVED** [review]
QA + 1

**Review by davidvarga93 — APPROVED** [review]
QA + 1

**Review by idevainstr — DISMISSED** [review]
QA+1 iPad 15.4

**Review by idevainstr — APPROVED** [review]

**Pull request merged**